src/main/java/org/broadinstitute/consent/http/health/SendGridHealthCheck.java

```diff
@@ -0,0 +1,68 @@
+package org.broadinstitute.consent.http.health;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.google.api.client.http.HttpStatusCodes;
+import com.google.gson.Gson;
+import io.dropwizard.lifecycle.Managed;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.broadinstitute.consent.http.configurations.MailConfiguration;
+import org.broadinstitute.consent.http.util.HttpClientUtil;
+
+import java.nio.charset.Charset;
+
+public class SendGridHealthCheck extends HealthCheck implements Managed {
+    private final HttpClientUtil clientUtil;
+    private final MailConfiguration configuration;
+
+    public SendGridHealthCheck(HttpClientUtil clientUtil, MailConfiguration configuration) {
+        this.clientUtil = clientUtil;
+        this.configuration = configuration;
+    }
+
+    @Override
+    protected Result check() throws Exception {
+        try {
+            String statusUrl = configuration.getSendGridStatusUrl();
+            HttpGet httpGet = new HttpGet(statusUrl);
+            try (CloseableHttpResponse response = clientUtil.getHttpResponse(httpGet)) {
+                if (response.getStatusLine().getStatusCode() == HttpStatusCodes.STATUS_CODE_OK) {
+                    String content = IOUtils.toString(response.getEntity().getContent(), Charset.defaultCharset());
+                    SendGridStatus sgStatus = new Gson().fromJson(content, SendGridStatus.class);
+                    if (sgStatus.status.description.equalsIgnoreCase("All Systems Operational")) {
+                        return Result.builder()
+                                .withDetail("pages", sgStatus.page)
+                                .withDetail("status", sgStatus.status)
+                                .healthy()
+                                .build();
+                    } else {
+                        return Result.unhealthy("SendGrid status is unhealthy: " + sgStatus.status.description);
+                    }
+                } else {
+                    return Result.unhealthy("SendGrid status is unhealthy: " + response.getStatusLine());
+                }
+            } catch (Exception e) {
+                return Result.unhealthy(e);
+            }
+        } catch (Exception e) {
+            return Result.unhealthy(e);
+        }
```

Comment:
I realize that you're copying the pattern from the Sam check here, but we should strive to reduce the nested try/catch blocks. This could be condensed down with some pre-checks on values to ensure that we are throwing the right error in the right place.

Reply:
I'm changing my mind on this ... I think we really do need the nested try/catch because of the HttpGet's attempt to coerce the status url to a URI.